portal/static/js/src/components/LongitudinalReport.vue

```diff
@@ -0,0 +1,335 @@
+            this.domains = this.questionnaireData.item.filter(item => {
+                    return item.code && item.code.length;
+            }).map(item => {
+                return item.code[0].code;
+            });
+            //de-duplicate
+            this.domains = this.domains.filter((d, index) => {
+                return this.domains.indexOf(d) === index;
+            });
+        },
+        setQuestions() {
+            if (!this.questionnaireData) {
+                return;
+            }
+            this.questions = this.questionnaireData.item;
+        },
+        setQuestionnaireDates() {
+            /*
+             * available response authored dates
+             */
+            this.questionnaireDates = this.assessmentData.map(item => {
+                return this.setDisplayDate(item.authored);
+            });
+        },
+        setDisplayData() {
+            /*
+             * prepping data for display
+             */
+            this.assessmentData.forEach((item, index) => {
+                this.data[index] = {
+                    "date": this.setDisplayDate(item.authored),
+                    "data": []};
+                (item.group.question).forEach(entry => {
+                    var q = (entry.text ? entry.text.replace(/^[\d\w]{1,3}\./, "") : ""),
+                        a = "";
+                    entry.answer = entry.answer || [];
+                    var arrValueStrings = $.grep(entry.answer, (item) => {
+                        return this.hasValue(item.valueString);
+                    });
+                    arrValueStrings = arrValueStrings.map(function(item) {
+                        return item.valueString;
+                    });
+                    a = arrValueStrings.join("<br/>");
+
+                    var arrValueCoding = $.grep(entry.answer, function(item) {
+                        return item.valueCoding && item.valueCoding.code;
+                    });
+                    arrValueCoding = arrValueCoding.map(function(item) {
+                        return item.valueCoding.code;
+                    });
+                    /*
+                    * using valueCoding.code for answer and linkId for question if BOTH question and answer are empty strings
+                    */
+                    if (!q && !a) {
+                        q = entry.linkId;
+                        a = arrValueCoding.join("<br/>");
+                    }
+                    q = !this.hasValue(q) ? "--" : q;
+                    a = !this.hasValue(a) ? "--" : a;
+                    let answerValue = arrValueCoding.length? parseInt(arrValueCoding[0].split(".")[2]) : 0;
+                    let answerObj = {
+                        q: q,
+                        a: a,
+                        linkId: entry.linkId,
+                        value: answerValue,
+                        cssClass: answerValue >= 4 ? "darkest" : (answerValue >= 3 ? "darker": (answerValue <= 1 ? "no-value": ""))
```

Comment:
if darkest and darker correspond with the 2 worst values for a question, this may not work for the `social_isolation` question which only has 4 options (vs 5 for the others) [as Justin mentioned here](https://github.com/uwcirg/truenth-portal/pull/3814#discussion_r512348087)

Reply:
thank you, Ivan, for the heads-up! I think I fixed it.